scripts/biltrans-extract-frac-freq.py

```diff
@@ -17,32 +17,40 @@
 #
 #
 
-# The sl-tl possible combinations
-sl_tl = defaultdict(lambda: defaultdict(lambda: 0.0))
-
-
-class Counter(BCC.BiltransCounter):
-    tokenizer = 'biltrans'
-    line_ids = True
-
-    def process_lu(self, sl, tl, idx, cur_sl_row, frac_count=0):
-        global sl_tl
-        sl_tl[sl][tl] += frac_count
-
-
-c = Counter()
-c.read_files(sys.argv[1],  # File with ambiguous biltrans output
-             sys.argv[2])  # File with disambiguated biltrans output
-
-for sl in sl_tl:
-    newtl = sorted(sl_tl[sl], key=lambda x: sl_tl[sl][x])
-    newtl.reverse()
-    first = True
-    for tl in newtl:
-        if first:
-            print('%.10f %s %s @' %
-                  (sl_tl[sl][tl], common.wrap(sl), common.wrap(tl)))
-            first = False
-        else:
-            print('%.10f %s %s' %
-                  (sl_tl[sl][tl], common.wrap(sl), common.wrap(tl)))
+def biltrans_extract_frac_freq(biltrans_ambig, biltrans_annotated):
+    # The sl-tl possible combinations
+    sl_tl = defaultdict(lambda: defaultdict(lambda: 0.0))
+
+
+    class Counter(BCC.BiltransCounter):
+        tokenizer = 'biltrans'
+        line_ids = True
+
+        def process_lu(self, sl, tl, idx, cur_sl_row, frac_count=0):
+            global sl_tl
```

Comment:
I'm pretty sure this line will break because `sl_tl` isn't a global variable, so this should be `nonlocal sl_tl`.

Alternatively, I think the more normal Python approach would be to have the class in global scope and have `sl_tl` be a variable within the class.

Reply:
updated